FIX review csubs cache devel doc + hardening/fixes in csub cache (#3987)
telefonicaid/fiware-orion: doc/manuals/devel/mongoBackend.md

* `mongoUpdateSubscription()` is invoked from a service routine (step 1). This can be from either `patchSubscription()` (which resides in `lib/serviceRoutinesV2/patchSubscription.cpp`) or `mongoUpdateContextSubscription()` (which resides in `lib/mongoBackend/mongoUpdateContextSubscription.cpp`).
* Depending on `-reqMutexPolicy`, the request semaphore may be taken (write mode) (step 2). See [this document for details](semaphores.md#mongo-request-semaphore). 

> * An update request is build using MongoDB `$set`/`$unset` operators, then used to update the subscription in DB using `colletionFindAndModify()` in the `connectionOperations` module (steps 3 and 4).

Reviewer: I think this sentence has to be rephrased, it is not clear what is done in this step
```suggestion
The subscription is updated in DB by a request using MongoDB `$set`/`$unset` operators. This operation is done in the function `colletionFindAndModify()` in the `connectionOperations` module (steps 3 and 4).

```

Author: ```suggestion
* The subscription is updated in DB using MongoDB `$set`/`$unset` operators. This operation is done in the function `colletionFindAndModify()` in the `connectionOperations` module (steps 3 and 4).
```

Author: Fixed in b4ee08b
